src/cmd/transpile.js

```diff
@@ -14,6 +14,22 @@ import { platform } from 'node:process';
 
 const isWindows = platform === 'win32';
 
+const DEFAULT_ASYNC_IMPORTS = [
+  "wasi:io/poll#poll",
+  "wasi:io/poll#[method]pollable.block",
+  "wasi:io/streams#[method]input-stream.blocking-read",
+  "wasi:io/streams#[method]input-stream.blocking-skip",
+  "wasi:io/streams#[method]output-stream.blocking-flush",
+  "wasi:io/streams#[method]output-stream.blocking-write-and-flush",
+  "wasi:io/streams#[method]output-stream.blocking-write-zeroes-and-flush",
+  "wasi:io/streams#[method]output-stream.blocking-splice",
+];
```

Comment:
It could be nice to automatically include a preview2 implementation designed for async now, and run the preview2 tests against that.

One issue here is how to get the async imports, but an easy option might be to just use a convention like:

```js
import ... from '@bytecodealliance/preview2-shims/async/io'`;
```

This would be amazing to explore further...

Reply:
Yep, I have a working draft of WASI P2 shims that I have been testing against. But was thinking of breaking into a separate PR.